### PR TITLE
fix issue #808 - only show find in page on websites, hide on top sites

### PR DESCRIPTION
**Pull request by jamesmudgett**

### CODE REVIEW

File: Client/Frontend/Browser/BrowserViewController/BrowserViewController.swift [hunk 1]
```diff
@@ -1003,6 +1003,19 @@ extension BrowserViewController: SearchViewControllerDelegate {
         updateFindInPageVisibility(visible: true)
         findInPageBar?.text = query
     }
+    
+    func searchViewControllerAllowFindInPage() -> Bool {
+        // Hides find in page for new tabs.
+        
+        if let st = tabManager.selectedTab {
```
Comment (jhreis) [inline Client/Frontend/Browser/BrowserViewController/BrowserViewController.swift]: You can chain optional unwrappings, and simplify the return here:
```
if let st = tabManager.selectedTab, let wv = st.webView {
    return !AboutUtils.isAboutHomeURL(wv.URL)
}
return false
```

File: Client/Frontend/Browser/SearchViewController.swift [hunk 2]
```diff
@@ -509,9 +510,14 @@ extension SearchViewController {
         case .SearchSuggestions:
             return 0
         case .BookmarksAndHistory:
-            return 0
+            return 1
         case .FindInPage:
-            return 22
+            if let sd = searchDelegate {
```
Comment (jhreis) [inline Client/Frontend/Browser/SearchViewController.swift]: Can use `where`, to remove the ternary and duplicate `0` return:
```
if let sd = searchDelegate where sd.searchViewControllerAllowFindInPage() {
    return 22
}
return 0
```

File: Client/Frontend/Browser/SearchViewController.swift [hunk 3]
```diff
@@ -565,7 +575,12 @@ extension SearchViewController {
         case .BookmarksAndHistory:
             return data.count
         case .FindInPage:
-            return 1
+            if let sd = searchDelegate {
```
Comment (jhreis) [inline Client/Frontend/Browser/SearchViewController.swift]: Same as above, using `where`

File: Client/Frontend/Browser/SearchViewController.swift (hunk 2 above)
Comment (jamesmudgett) [inline Client/Frontend/Browser/SearchViewController.swift]: Great! Will update

**Pull request merged**